airflow/models/mappedoperator.py

```diff
@@ -284,7 +285,10 @@ def get_serialized_fields(cls):
     @staticmethod
     @cache
     def deps_for(operator_class: Type["BaseOperator"]) -> FrozenSet[BaseTIDep]:
-        return operator_class.deps | {MappedTaskIsExpanded()}
+        operator_deps = operator_class.deps
+        if not isinstance(operator_deps, collections.abc.Set):
+            raise UnmappableOperator(f"cannot map operator; f{operator_class}.deps must be a set")
```

Comment:
Can we be more explicit here -- if it is defined as a property to tell the user what is wrong as this could easily be confusing to the user if they have it as:

```python
    @property
    def deps(self):
       return {MyDep()}
```

Because at first glance this _is_ a set.

Reply:
Anything suggestion on wording? I’m struggling to come up with a succinct description.

Reply:
"deps must be defined as a class-level variable, not a property"?

Reply:
Updated